File: testers/grpc/java-spring/generated-and-checked-in/com/example/grpc/EchoServiceClient.java
package com.example.grpc;

import io.grpc.CallOptions;
import io.grpc.Channel;
import io.grpc.MethodDescriptor;
import io.grpc.MethodDescriptor.MethodType;
import io.grpc.stub.ClientCalls;

/** gRPC client wrapper for EchoService - wraps Channel with clean types */
public class EchoServiceClient implements EchoService {
  Channel channel;

  public EchoServiceClient(Channel channel) {
    this.channel = channel;
  }

  public static MethodDescriptor<Customer, Customer> ECHO_CUSTOMER =
      MethodDescriptor.newBuilder(Customer.MARSHALLER, Customer.MARSHALLER)
          .setType(MethodType.UNARY)
          .setFullMethodName("testgrpc.EchoService/EchoCustomer")
          .build();

  public static MethodDescriptor<Inventory, Inventory> ECHO_INVENTORY =
      MethodDescriptor.newBuilder(Inventory.MARSHALLER, Inventory.MARSHALLER)
          .setType(MethodType.UNARY)
          .setFullMethodName("testgrpc.EchoService/EchoInventory")
          .build();

  public static MethodDescriptor<Notification, Notification> ECHO_NOTIFICATION =
      MethodDescriptor.newBuilder(Notification.MARSHALLER, Notification.MARSHALLER)
          .setType(MethodType.UNARY)
          .setFullMethodName("testgrpc.EchoService/EchoNotification")
          .build();

  public static MethodDescriptor<OptionalFields, OptionalFields> ECHO_OPTIONAL_FIELDS =
      MethodDescriptor.newBuilder(OptionalFields.MARSHALLER, OptionalFields.MARSHALLER)
          .setType(MethodType.UNARY)
          .setFullMethodName("testgrpc.EchoService/EchoOptionalFields")
          .build();

  public static MethodDescriptor<Order, Order> ECHO_ORDER =
      MethodDescriptor.newBuilder(Order.MARSHALLER, Order.MARSHALLER)
          .setType(MethodType.UNARY)
          .setFullMethodName("testgrpc.EchoService/EchoOrder")
          .build();

  public static MethodDescriptor<Outer, Outer> ECHO_OUTER =
      MethodDescriptor.newBuilder(Outer.MARSHALLER, Outer.MARSHALLER)
          .setType(MethodType.UNARY)
          .setFullMethodName("testgrpc.EchoService/EchoOuter")
          .build();

  public static MethodDescriptor<PaymentMethod, PaymentMethod> ECHO_PAYMENT_METHOD =
      MethodDescriptor.newBuilder(PaymentMethod.MARSHALLER, PaymentMethod.MARSHALLER)
          .setType(MethodType.UNARY)
          .setFullMethodName("testgrpc.EchoService/EchoPaymentMethod")
          .build();

  public static MethodDescriptor<ScalarTypes, ScalarTypes> ECHO_SCALAR_TYPES =
      MethodDescriptor.newBuilder(ScalarTypes.MARSHALLER, ScalarTypes.MARSHALLER)
          .setType(MethodType.UNARY)
          .setFullMethodName("testgrpc.EchoService/EchoScalarTypes")
          .build();

  public static MethodDescriptor<WellKnownTypesMessage, WellKnownTypesMessage>
      ECHO_WELL_KNOWN_TYPES =
          MethodDescriptor.newBuilder(
                  WellKnownTypesMessage.MARSHALLER, WellKnownTypesMessage.MARSHALLER)
              .setType(MethodType.UNARY)
              .setFullMethodName("testgrpc.EchoService/EchoWellKnownTypes")
              .build();

  @Override
  public ScalarTypes echoScalarTypes(ScalarTypes request) {
    return ClientCalls.blockingUnaryCall(
        channel, EchoServiceClient.ECHO_SCALAR_TYPES, CallOptions.DEFAULT, request);
  }

  @Override
  public Customer echoCustomer(Customer request) {
    return ClientCalls.blockingUnaryCall(
        channel, EchoServiceClient.ECHO_CUSTOMER, CallOptions.DEFAULT, request);
  }

  @Override
  public Order echoOrder(Order request) {
    return ClientCalls.blockingUnaryCall(
        channel, EchoServiceClient.ECHO_ORDER, CallOptions.DEFAULT, request);
  }

  @Override
  public Inventory echoInventory(Inventory request) {
    return ClientCalls.blockingUnaryCall(
        channel, EchoServiceClient.ECHO_INVENTORY, CallOptions.DEFAULT, request);
  }

  @Override
  public Outer echoOuter(Outer request) {
    return ClientCalls.blockingUnaryCall(
        channel, EchoServiceClient.ECHO_OUTER, CallOptions.DEFAULT, request);
  }

  @Override
  public OptionalFields echoOptionalFields(OptionalFields request) {
    return ClientCalls.blockingUnaryCall(
        channel, EchoServiceClient.ECHO_OPTIONAL_FIELDS, CallOptions.DEFAULT, request);
  }

  @Override
  public WellKnownTypesMessage echoWellKnownTypes(WellKnownTypesMessage request) {
    return ClientCalls.blockingUnaryCall(
        channel, EchoServiceClient.ECHO_WELL_KNOWN_TYPES, CallOptions.DEFAULT, request);
  }

  @Override
  public PaymentMethod echoPaymentMethod(PaymentMethod request) {
    return ClientCalls.blockingUnaryCall(
        channel, EchoServiceClient.ECHO_PAYMENT_METHOD, CallOptions.DEFAULT, request);
  }

  @Override
  public Notification echoNotification(Notification request) {
    return ClientCalls.blockingUnaryCall(
        channel, EchoServiceClient.ECHO_NOTIFICATION, CallOptions.DEFAULT, request);
  }
}

File: testers/grpc/scala-cats/generated-and-checked-in/com/example/grpc/EchoServiceClient.scala
package com.example.grpc

import cats.effect.IO
import io.grpc.CallOptions
import io.grpc.Channel
import io.grpc.MethodDescriptor
import io.grpc.MethodDescriptor.MethodType
import io.grpc.stub.ClientCalls

/** gRPC client wrapper for EchoService - wraps Channel with clean types */
class EchoServiceClient(val channel: Channel) extends EchoService {
  override def echoScalarTypes(request: ScalarTypes): IO[ScalarTypes] = {
    return IO.delay((() => ClientCalls.blockingUnaryCall(channel, EchoServiceClient.ECHO_SCALAR_TYPES, CallOptions.DEFAULT, request)).apply())
  }

  override def echoCustomer(request: Customer): IO[Customer] = {
    return IO.delay((() => ClientCalls.blockingUnaryCall(channel, EchoServiceClient.ECHO_CUSTOMER, CallOptions.DEFAULT, request)).apply())
  }

  override def echoOrder(request: Order): IO[Order] = {
    return IO.delay((() => ClientCalls.blockingUnaryCall(channel, EchoServiceClient.ECHO_ORDER, CallOptions.DEFAULT, request)).apply())
  }

  override def echoInventory(request: Inventory): IO[Inventory] = {
    return IO.delay((() => ClientCalls.blockingUnaryCall(channel, EchoServiceClient.ECHO_INVENTORY, CallOptions.DEFAULT, request)).apply())
  }

  override def echoOuter(request: Outer): IO[Outer] = {
    return IO.delay((() => ClientCalls.blockingUnaryCall(channel, EchoServiceClient.ECHO_OUTER, CallOptions.DEFAULT, request)).apply())
  }

  override def echoOptionalFields(request: OptionalFields): IO[OptionalFields] = {
    return IO.delay((() => ClientCalls.blockingUnaryCall(channel, EchoServiceClient.ECHO_OPTIONAL_FIELDS, CallOptions.DEFAULT, request)).apply())
  }

  override def echoWellKnownTypes(request: WellKnownTypesMessage): IO[WellKnownTypesMessage] = {
    return IO.delay((() => ClientCalls.blockingUnaryCall(channel, EchoServiceClient.ECHO_WELL_KNOWN_TYPES, CallOptions.DEFAULT, request)).apply())
  }

  override def echoPaymentMethod(request: PaymentMethod): IO[PaymentMethod] = {
    return IO.delay((() => ClientCalls.blockingUnaryCall(channel, EchoServiceClient.ECHO_PAYMENT_METHOD, CallOptions.DEFAULT, request)).apply())
  }

  override def echoNotification(request: Notification): IO[Notification] = {
    return IO.delay((() => ClientCalls.blockingUnaryCall(channel, EchoServiceClient.ECHO_NOTIFICATION, CallOptions.DEFAULT, request)).apply())
  }
}

object EchoServiceClient {
  val ECHO_CUSTOMER: MethodDescriptor[Customer, Customer] = MethodDescriptor.newBuilder(Customer.marshaller, Customer.marshaller).setType(MethodType.UNARY).setFullMethodName("testgrpc.EchoService/EchoCustomer").build()

  val ECHO_INVENTORY: MethodDescriptor[Inventory, Inventory] = MethodDescriptor.newBuilder(Inventory.marshaller, Inventory.marshaller).setType(MethodType.UNARY).setFullMethodName("testgrpc.EchoService/EchoInventory").build()

  val ECHO_NOTIFICATION: MethodDescriptor[Notification, Notification] = MethodDescriptor.newBuilder(Notification.marshaller, Notification.marshaller).setType(MethodType.UNARY).setFullMethodName("testgrpc.EchoService/EchoNotification").build()

  val ECHO_OPTIONAL_FIELDS: MethodDescriptor[OptionalFields, OptionalFields] = MethodDescriptor.newBuilder(OptionalFields.marshaller, OptionalFields.marshaller).setType(MethodType.UNARY).setFullMethodName("testgrpc.EchoService/EchoOptionalFields").build()

  val ECHO_ORDER: MethodDescriptor[Order, Order] = MethodDescriptor.newBuilder(Order.marshaller, Order.marshaller).setType(MethodType.UNARY).setFullMethodName("testgrpc.EchoService/EchoOrder").build()

  val ECHO_OUTER: MethodDescriptor[Outer, Outer] = MethodDescriptor.newBuilder(Outer.marshaller, Outer.marshaller).setType(MethodType.UNARY).setFullMethodName("testgrpc.EchoService/EchoOuter").build()

  val ECHO_PAYMENT_METHOD: MethodDescriptor[PaymentMethod, PaymentMethod] = MethodDescriptor.newBuilder(PaymentMethod.marshaller, PaymentMethod.marshaller).setType(MethodType.UNARY).setFullMethodName("testgrpc.EchoService/EchoPaymentMethod").build()

  val ECHO_SCALAR_TYPES: MethodDescriptor[ScalarTypes, ScalarTypes] = MethodDescriptor.newBuilder(ScalarTypes.marshaller, ScalarTypes.marshaller).setType(MethodType.UNARY).setFullMethodName("testgrpc.EchoService/EchoScalarTypes").build()

  val ECHO_WELL_KNOWN_TYPES: MethodDescriptor[WellKnownTypesMessage, WellKnownTypesMessage] = MethodDescriptor.newBuilder(WellKnownTypesMessage.marshaller, WellKnownTypesMessage.marshaller).setType(MethodType.UNARY).setFullMethodName("testgrpc.EchoService/EchoWellKnownTypes").build()
}